3.analyze_data/utils/isolation_forest_data_feature_importance.py

```diff
@@ -0,0 +1,148 @@
+from collections import defaultdict
+from typing import Optional, Sequence, Union
+
+import numpy as np
+import pandas as pd
+from joblib import Parallel, delayed
+from sklearn.tree import DecisionTreeRegressor, _tree
+
+
+class IsoforestFeatureImportance:
+    # Computes Isolation Forest (Scikit-learn) feature importances from a subset of data.
+
+    def __init__(
+        self,
+        _estimators: Sequence[DecisionTreeRegressor],
+        _morphology_data: pd.DataFrame,
+    ):
+        """
+        Parameters
+        ----------
+        _estimators: Scikit-learn decision tree regressor estimators.
+        _morphology_data: Data with only numerical morphology data (without metadata).
+        """
+
+        self._estimators = _estimators
+        self._morphology_data = _morphology_data
+        self._isoforest_importances = None
+
+    @property
+    def isoforest_importances(self):
+        if self._isoforest_importances is None:
+            raise ValueError("isoforest_importances have not been computed")
+
+        return self._isoforest_importances
```

Comment:
I'm a bit confused, but probably b/c I don't understand properties too well, but why is it returning itself?